src/dynarec/arm64/dynarec_arm64_67.c

```diff
@@ -973,51 +957,31 @@ uintptr_t dynarec64_67(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
                 case 4:
                 case 6:
                     INST_NAME("SHL Ed, Ib");
-                    u8 = geted_ib(dyn, addr, ninst, nextop) & (rex.w ? 0x3f : 0x1f);
-                    // flags are not affected if count is 0, we make it a nop if possible.
-                    if (u8) {
-                        SETFLAGS(X_ALL, SF_SET_PENDING); // some flags are left undefined
-                        GETED32(1);
-                    } else
-                        FAKEED;
-                    F8;
+                    SETFLAGS(X_ALL, SF_SET_PENDING);    // some flags are left undefined
+                    GETED32(1);
+                    u8 = (F8)&(rex.w?0x3f:0x1f);
                     emit_shl32c(dyn, ninst, rex, ed, u8, x3, x4);
-                    if (u8) {
-                        WBACK;
-                    } else if (MODREG && !rex.w)
-                        MOVw_REG(ed, ed);
+                    WBACK;
                     break;
                 case 5:
                     INST_NAME("SHR Ed, Ib");
-                    u8 = geted_ib(dyn, addr, ninst, nextop) & (rex.w ? 0x3f : 0x1f);
-                    // flags are not affected if count is 0, we make it a nop if possible.
-                    if (u8) {
-                        SETFLAGS(X_ALL, SF_SET_PENDING); // some flags are left undefined
-                        GETED32(1);
-                    } else
-                        FAKEED;
-                    F8;
+                    SETFLAGS(X_ALL, SF_SET_PENDING);    // some flags are left undefined
+                    GETED32(1);
+                    u8 = (F8)&(rex.w?0x3f:0x1f);
                     emit_shr32c(dyn, ninst, rex, ed, u8, x3, x4);
-                    if (u8) {
+                    if(u8) {
                         WBACK;
-                    } else if (MODREG && !rex.w)
-                        MOVw_REG(ed, ed);
+                    }
                     break;
                 case 7:
                     INST_NAME("SAR Ed, Ib");
-                    u8 = geted_ib(dyn, addr, ninst, nextop) & (rex.w ? 0x3f : 0x1f);
-                    // flags are not affected if count is 0, we make it a nop if possible.
-                    if (u8) {
-                        SETFLAGS(X_ALL, SF_SET_PENDING); // some flags are left undefined
-                        GETED32(1);
-                    } else
-                        FAKEED;
-                    F8;
+                    SETFLAGS(X_ALL, SF_SET_PENDING);    // some flags are left undefined
+                    GETED32(1);
+                    u8 = (F8)&(rex.w?0x3f:0x1f);
                     emit_sar32c(dyn, ninst, rex, ed, u8, x3, x4);
-                    if (u8) {
+                    if(u8) {
                         WBACK;
-                    } else if (MODREG && !rex.w)

```

Comment:
why removing this part?

Reply:
Ah, it was not there before, I see

Reply:
Yeah, I reverted the changes made to arm64.